docs/src/main/asciidoc/security-proactive-authentication-concept.adoc

```diff
@@ -0,0 +1,151 @@
+////
+This document is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+[id="security-proactive-authentication-concept"]
+= Proactive authentication
+include::_attributes.adoc[]
+:categories: security,web
+
+Proactive authentication is enabled in Quarkus by default. This means that if an incoming request has a credential then that request will always be authenticated, even if the target page does not require authentication.
+
+[[proactive-authentication]]
+
+Requests with an invalid credential will always be rejected, even when the page is public. 
+
+If you only want to authenticate when the target page requires authentication, you can change the default behavior.
+
+To disable proactive authentication in Quarkus, set the following attribute in the `application.properties` configuration file: 
+
+[source,xml,options="nowrap",role="white-space-pre"]
+----
+`quarkus.http.auth.proactive=false` 
+----
+
+If you disable proactive authentication then the authentication process will only be run when an identity is requested, either because there are security rules that require the user to be authenticated, or due to programmatic access to the current identity.
+
+If proactive authentication is in use, accessing the `SecurityIdentity` is a blocking operation. 
+This is because authentication may not have happened yet, and accessing it may require calls to external systems such as databases that may block.
```

Comment:
```suggestion
This is because authentication may have yet to happen, and accessing it may require calls to external systems, such as databases that may block.
```